apps/token-manager/app/src/script.js

```diff
@@ -47,67 +44,66 @@ retryEvery(retry => {
   })
 })
 
-async function initialize(tokenAddr) {
-  const token = app.external(tokenAddr, tokenAbi)
-  try {
-    const tokenSymbol = await token.symbol().toPromise()
-    app.identify(tokenSymbol)
-  } catch (err) {
-    console.error(
-      `Failed to load token symbol for token at ${tokenAddr} due to:`,
-      err
-    )
-  }
-
-  return createStore(token, tokenAddr)
-}
+async function initialize(tokenAddress) {
+  const token = app.external(tokenAddress, tokenAbi)
 
-// Hook up the script as an aragon.js store
-async function createStore(token, tokenAddr) {
   return app.store(
     async (state, { address, event, returnValues }) => {
       let nextState = {
         ...state,
-        // Fetch the app's settings, if we haven't already
-        ...(!hasLoadedTokenSettings(state)
-          ? await loadTokenSettings(token)
-          : {}),
       }
 
-      if (event === INITIALIZATION_TRIGGER) {
-        nextState = {
-          ...nextState,
-          tokenAddress: tokenAddr,
-          maxAccountTokens: await app.call('maxAccountTokens').toPromise(),
-        }
-      } else if (addressesEqual(address, tokenAddr)) {
+      if (addressesEqual(address, tokenAddress)) {
         switch (event) {
           case 'ClaimedTokens':
-            if (addressesEqual(returnValues._token, tokenAddr)) {
-              nextState = await claimedTokens(token, nextState, returnValues)
+            if (addressesEqual(returnValues._token, tokenAddress)) {
+              return claimedTokens(token, nextState, returnValues)
             }
-            break
+            return nextState
           case 'Transfer':
-            nextState = await transfer(token, nextState, returnValues)
-            break
+            return transfer(token, nextState, returnValues)
           default:
-            break
+            return nextState
         }
-      } else {
-        // TODO: add handlers for the vesting events from token Manager
       }
+      // TODO: add handlers for the vesting events from token Manager
 
       return nextState
     },
-    [
-      // Always initialize the store with our own home-made event
-      of({ event: INITIALIZATION_TRIGGER }),
-      // Merge in the token's events into the app's own events for the store
-      token.events(),
-    ]
+    {
+      externals: [{ contract: token }],
+      init: initState({ token, tokenAddress }),
+    }
   )
 }
 
+const initState = ({ token, tokenAddress }) => async cachedState => {
```

Comment:
Nitpick: we might as well use the normal function declaration since there's no need for the lexical binding, and all the other functions are declared using the function syntax.